src/pe_source/data/helpers/redact_pii.py

```diff
@@ -0,0 +1,560 @@
+"""Functions to redact PII from a dataframe."""
+
+# Standard Python Libraries
+import re
+
+# Third-Party Libraries
+from presidio_analyzer import AnalyzerEngine
+from presidio_anonymizer import AnonymizerEngine
+import scrubadub
+import scrubadub.detectors.date_of_birth
+
+FL = [
+    r"(?:(?<=\s)|(?<=^))[a-zA-Z] \d{3} \d{3} \d{3} \d{3}(?=$|\s)",
+    r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{3}-\d{3}-\d{2}-\d{3}-\d(?=$|\s)",
+    r"(?:(?<=\s)|(?<=^))[a-zA-Z]-\d{3}-\d{3}-\d{3}-\d{3}(?=$|\s)",
+]
+HI_NE_VA = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{8}(?=$|\s)"]
+IL = [
+    r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{3}-\d{4}-\d{4}(?=$|\s)",
+    r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{11}(?=$|\s)",
+]
+MN_FL_MD_MI = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{12}(?=$|\s)"]
+MO_OK = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{9}(?=$|\s)"]
+MD = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]-\d{3}-\d{3}-\d{3}-\d{3}(?=$|\s)"]
+
+CA = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{7}(?=$|\s)"]
+CO = [r"(?:(?<=\s)|(?<=^))\d{2}-\d{3}-\d{4}(?=$|\s)"]
+ID = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]{2}\d{6}[a-zA-Z](?=$|\s)"]
+IO = [r"(?:(?<=\s)|(?<=^))\d{4}-\d{2}-\d{4}(?=$|\s)"]
+IA = [r"(?:(?<=\s)|(?<=^))\d{3}[a-zA-Z]{2}\d{4}(?=$|\s)"]
+KS = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{2}-\d{2}-\d{4}(?=$|\s)"]
+KY = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{2}-\d{3}-\d{3}(?=$|\s)"]
+MI = [r"(?:(?<=\s)|(?<=^))[a-zA-Z] \d{3} \d{3} \d{3} \d{3}(?=$|\s)"]
+NH = [
+    r"(?:(?<=\s)|(?<=^))([0][1-9]|[1][0-2])[a-zA-Z]{3}\d{2}(0[1-9]|[1-2][0-9]|3[0-1])\d(?=$|\s)"
+]
+NJ = [
+    r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{4}-\d{5}-\d{5}(?=$|\s)",
+    r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{14}(?=$|\s)",
+]
+NY = [r"(?:(?<=\s)|(?<=^))\d{3} \d{3} \d{3}(?=$|\s)"]
+ND = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]{3}-\d{2}-\d{4}(?=$|\s)"]
+OH = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]{3}-\d{2}-\d{4}(?=$|\s)"]
+PA = [r"(?:(?<=\s)|(?<=^))\d{2} \d{3} \d{3}(?=$|\s)"]
+VT = [r"(?:(?<=\s)|(?<=^))\d{7}[a-zA-Z](?=$|\s)"]
+VA = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{2}-\d{2}-\d{4}(?=$|\s)"]
+WA = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]{3}\*\*[a-zA-Z]{2}\d{3}[a-zA-Z]\d(?=$|\s)"]
+WV = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{6}(?=$|\s)"]
+WI = [r"(?:(?<=\s)|(?<=^))[a-zA-Z]\d{3}-\d{4}-\d{4}-\d{2}(?=$|\s)"]
+WY = [r"(?:(?<=\s)|(?<=^))\d{6}-\d{3}(?=$|\s)"]
+
+
+# Build detectors to find Drivers License ID
+
+
+class FL_DLFilth(scrubadub.filth.Filth):
+    """Create filth class for FL drivers licenses."""
+
+    type = "FL_drivers_license"
+
+
+class FL_DLDetector(scrubadub.detectors.RegexDetector):
+    """Create detector class to identify FL drivers licenses."""
+
+    name = "FL_drivers_license"
+    regex = re.compile("|".join(FL), re.IGNORECASE)
+    filth_cls = FL_DLFilth
+
+
+class HI_NE_VA_DLFilth(scrubadub.filth.Filth):
+    """Create filth class for HI and NE drivers licenses."""
+
+    type = "HI_NE_VA_drivers_license"
+
+
+class HI_NE_VA_DLDetector(scrubadub.detectors.RegexDetector):
+    """Create detector class to identify HI, NE and VA drivers licenses."""
```

Comment:
Oxford comma for the win! 😃 
```suggestion
    """Create detector class to identify HI, NE, and VA drivers licenses."""
```